src/components/text-editor/utils/html-converter.ts

```diff
@@ -0,0 +1,21 @@
+import { ContentTypeConverter } from './content-type-converter';
+import { EditorView } from 'prosemirror-view';
+
+/**
+ * @private
+ */
+export class HTMLConverter implements ContentTypeConverter {
+    public parseAsHTML = (text: string): Promise<string> => {
+        return new Promise((resolve) => {
+            resolve(text);
+        });
+    };
+
+    public serialize = (view: EditorView): string => {
+        if (view.dom.textContent === '') {
+            return '';
+        } else {
+            return view.dom.innerHTML;
```

Comment:
I don't think it's good practice to read directly from the `innerHTML` of prosemirror.